src/schedule/zephyr_ll.c

```diff
@@ -84,6 +84,34 @@ static void zephyr_ll_task_insert_unlocked(struct zephyr_ll *sch, struct task *t
 		list_item_append(&task->list, &sch->tasks);
 }
 
+/* perf measurement windows size 2^x */
+#define CYCLES_WINDOW_SIZE	10
+
+static inline enum task_state do_task_run(struct task *task)
+{
+	const int window_size = 10;
+	enum task_state state;
+	uint32_t cycles0, diff;
+
+	cycles0 = k_cycle_get_32();
+	state = task_run(task);
+	diff = k_cycle_get_32() - cycles0;
+
+	task->cycles_sum += diff;
+	task->cycles_max = diff > task->cycles_max ? diff : task->cycles_max;
+
+	if (++task->cycles_cnt == 1 << CYCLES_WINDOW_SIZE) {
+		task->cycles_sum >>= CYCLES_WINDOW_SIZE;
+		tr_info(&ll_tr, "ll task %p avg %lu, max %u",
```

Comment:
btw, can we get pipeline ID here instead of task pointer ?

Reply:
@lgirdwood That would be nicer, but not sure how we could get the pipeline ID here (that's audio specific concept, not visible at task level). I did add "task>uid" to the trace in V2 patch. That helps a bit.